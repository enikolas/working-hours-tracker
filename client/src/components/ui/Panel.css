.panel {
  display: flex;
  flex-direction: row;
  align-items: baseline;
  width: 100%;
  margin-bottom: 25px;
  border-radius: 6px;
  min-height: 25px;
}

.panel .icon {
  display: table-cell;
  float: left;
  align-self: center;
  vertical-align: middle;
  text-align: center;
  background-color: #fff;
  margin: 0 8px 0 14px;
  width: 30px;
  height: 27px;
  border-radius: 50%;
}

.panel .message {
  padding: 12px;
  font-size: 14px;
  width: calc(100% - 25px);
  color: #fff;
}

.panel.success {
  background-color: #479030;
}

.panel.success .icon:before {
  content: '✓';
  font-size: 22px;
  color: #479030;
}

.panel.error {
  background-color: #dc3545;
}

.panel.error .icon:before {
  content: '⚠';
  font-size: 22px;
  color: #dc3545;
}
