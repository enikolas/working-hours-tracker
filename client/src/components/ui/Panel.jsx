import React from 'react';
import PropTypes from 'prop-types';

import './Panel.css';

const Panel = ({ message, type }) => {
  if (message) {
    return (
      <div className={`panel ${type}`}>
        <div className="icon" />
        <div className="message">
          {message}
        </div>
      </div>
    );
  }

  return null;
};

export default Panel;

Panel.propTypes = {
  type: PropTypes.oneOf(['success', 'error']).isRequired,
  message: PropTypes.string
};

Panel.defaultProps = {
  message: null
};
