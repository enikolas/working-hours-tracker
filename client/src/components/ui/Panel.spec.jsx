import React from 'react';
import { shallow } from 'enzyme';

import Panel from './Panel';

describe('Panel', () => {
	it('should render properly with min props', () => {
		const wrapper = shallow(<Panel type="error" />);
		expect(wrapper).toMatchSnapshot();
	});
	it('should render with a message', () => {
		const wrapper = shallow(<Panel type="success" message="this is a panel message" />);
		expect(wrapper).toMatchSnapshot();
	});
});
