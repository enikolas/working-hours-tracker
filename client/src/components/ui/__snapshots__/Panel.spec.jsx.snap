// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Panel should render properly with min props 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Panel
    message={null}
    type="error"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): null,
  Symbol(enzyme.__nodes__): Array [
    null,
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`Panel should render with a message 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Panel
    message="this is a panel message"
    type="success"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="icon"
        />,
        <div
          className="message"
        >
          this is a panel message
        </div>,
      ],
      "className": "panel success",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "icon",
        },
        "ref": null,
        "rendered": null,
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "this is a panel message",
          "className": "message",
        },
        "ref": null,
        "rendered": "this is a panel message",
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="icon"
          />,
          <div
            className="message"
          >
            this is a panel message
          </div>,
        ],
        "className": "panel success",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "icon",
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "this is a panel message",
            "className": "message",
          },
          "ref": null,
          "rendered": "this is a panel message",
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
